### DISCUSSION TITLE
How do I use the iOS Simulator for a build?

### DISCUSSION THREAD

**Discussion by davewevans** [codemagic-ci-cd/codemagic-docs#2789] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Create simple FLutter app.
Build for iOS in Codemagic portal
Run app in iOS simulator

### Expected results

I have just signed up for the pay-as-you-go plan. I'm simply trying to build a Flutter app for iOS and run in a simultor. 
Here is the documentation about that: [Previewing iOS apps on a simulator](https://docs.codemagic.io/flutter-testing/app-preview/)
To run a build in a simulator, it says, "You can preview any .app artifact built in Codemagic that targets the iOS simulator. For such artifacts, there is a Quick Launch button available next to the artifact name on the build overview page."
I do not see a "Quick Launch button" any where. 
I would expect to see a "Quick Launch button" since that is what your documentation is telling me.

### Actual results

No Quick Launch button! I can't run this in an iOS simulator although your documentaion states that I can. I'm learning this after I entered my billing info. How do I run a Flutter app in an iOS simulator? Is that possible with Codemagic?

**Comment by icarusdust** [comment]
Hi @davewevans, you are probably missing --simulator argument to the build command. The produced binary needs to be a correct simulator package. So, go to apps settings and add --simulator argument in the build arguments in front of iOS. Refer to the screenshot if you are working with Workflow Editor. If working with `codemagic.yaml`, then just run the build command as `flutter build ios --debug --simulator`

<img width="908" alt="Screenshot 2024-10-18 at 15 03 44" src="https://github.com/user-attachments/assets/356006df-87df-4be8-a1df-48521a215006">

**Comment by icarusdust** [comment]
Closing the thread due to inactivity. You can either reply here or create a new thread.
